src/target/riscv/riscv-013.c

```diff
@@ -3749,33 +3783,14 @@ read_memory_abstract(struct target *target, const riscv_mem_access_args_t args)
 		/* Execute the command */
 		uint32_t cmderr;
 		result = riscv013_execute_abstract_command(target, command, &cmderr);
-
-		/* TODO: we need to modify error handling here. */
-		/* NOTE: in case of timeout cmderr is set to CMDERR_NONE */
-		if (info->has_aampostincrement == YNM_MAYBE) {
-			if (result == ERROR_OK) {
-				/* Safety: double-check that the address was really auto-incremented */
-				riscv_reg_t new_address;
-				result = read_abstract_arg(target, &new_address, 1, riscv_xlen(target));
-				if (result != ERROR_OK)
-					return mem_access_result(MEM_ACCESS_FAILED_DM_ACCESS_FAILED);
-
-				if (new_address == args.address + args.size) {
-					LOG_TARGET_DEBUG(target, "aampostincrement is supported on this target.");
-					info->has_aampostincrement = YNM_YES;
-				} else {
-					LOG_TARGET_WARNING(target, "Buggy aampostincrement! Address not incremented correctly.");
-					info->has_aampostincrement = YNM_NO;
-				}
-			} else {
-				/* Try the same access but with postincrement disabled. */
-				command = access_memory_command(target, false, width, false, false);
-				result = riscv013_execute_abstract_command(target, command, &cmderr);
-				if (result == ERROR_OK) {
-					LOG_TARGET_DEBUG(target, "aampostincrement is not supported on this target.");
-					info->has_aampostincrement = YNM_NO;
-				}
-			}
+		if (use_aampostincrement && result != ERROR_OK &&
+					cmderr == CMDERR_NOT_SUPPORTED) {
+			LOG_TARGET_DEBUG(target, "aampostincrement is not supported on this target.");
```

Comment:
The command can result in cmderr==UNSUPPORTED due to other reasons than `command.aampostincrement` - for example due to `command.aamvirtual`.

For that reason, please adjust the message. For example:

```suggestion
			LOG_TARGET_DEBUG(target, "Trying the same abstract memory write command, but without aampostincrement");
```

Reply:
Addressed